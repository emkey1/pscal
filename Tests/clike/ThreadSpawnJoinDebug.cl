void child1() {
    printf("child1 start\n");
    printf("child1 end\n");
}

void child2() {
    printf("child2 start\n");
    printf("child2 end\n");
}

int main() {
    int tid1;
    int tid2;
    printf("parent start\n");

    // Spawn and join first child
    tid1 = spawn child1();
    join tid1;
    printf("joined child1\n");

    // Spawn and join second child
    tid2 = spawn child2();
    join tid2;
    printf("joined child2\n");

    printf("parent end\n");
    return 0;
}
