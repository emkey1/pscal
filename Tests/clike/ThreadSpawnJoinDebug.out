parent start
child1 start
child1 end
joined child1
child2 start
child2 end
joined child2
parent end
